### PR TITLE
adds clarifications on plugin use

### PR DESCRIPTION
**Pull request by d2r**
See #1737

### CODE REVIEW

**Comment by d2r** [comment]
Note, I was unable to build the documentation myself in a macOS environment.

I got unrelated errors like the following:
```
writing output... [  0%] admin-guide/configuration/cache-basics.en
Exception occurred:
  File "/usr/local/Cellar/sphinx-doc/1.6.2/libexec/lib/python2.7/site-packages/sphinx/util/osutil.py", line 61, in relative_uri
    if to.startswith(SEP):
UndefinedError: 'str object' has no attribute 'filename'
The full traceback has been saved in /var/folders/vt/83hvhtyn5k57dx_bjqy3130w0000gs/T/sphinx-err-xtThJm.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [html-local] Error 1
```

File: doc/admin-guide/plugins/header_rewrite.en.rst [hunk 1]
```diff
@@ -102,6 +102,9 @@ As with the global method above, multiple configuration files may be listed,
 each with its own ``@pparam=<file>`` and their contents will be evaluated in
 the order the files were specified.
 
+Each ruleset that is configured per-mapping must have a special
+`Hook Conditions`_ defined.
```
Comment (mlibbey) [inline doc/admin-guide/plugins/header_rewrite.en.rst]: Not strictly true, as rules without a Hook specified use the REMAP_PSEUDO_HOOK.

How about changing the "must" to "should", and adding, " Without a defined hook, these rulesets will use the REMAP_PSEUDO_HOOK."

Comment (d2r) [inline doc/admin-guide/plugins/header_rewrite.en.rst]: Thanks, will update.

**Comment by SolidWallOfCode** [comment]
[approve ci]

**Comment by d2r** [comment]
> Squash needed.

squashed

**Comment by SolidWallOfCode** [comment]
[approve ci]

**Comment by SolidWallOfCode** [comment]
[approve ci autest]

**Comment by d2r** [comment]
@SolidWallOfCode We can probably close #1737 now.

**Comment by mlibbey** [comment]
cherry-picked to 7.1.x

**Pull request merged**